drivers/modem/alt1250/alt1250.c

```diff
@@ -0,0 +1,1279 @@
+    {
+      /* Change the command ID to Version 1
+       * Even if it cannot be converted, try to search the table
+       * using the original command ID.
+       */
+
+      cidv1 = convert_cid2v1(cid);
+      if (cidv1 != APICMDID_UNKNOWN)
+        {
+          cid = cidv1;
+        }
+    }
+
+  return cid;
+}
+
+/****************************************************************************
+ * Name: get_bitmap
+ ****************************************************************************/
+
+static uint64_t get_bitmap(FAR struct alt1250_dev_s *dev, uint16_t cid,
+  uint8_t altver)
+{
+  uint64_t bitmap = 0ULL;
+
+  cid = cid_to_searchable(cid, altver);
+
+  search_evtbufinst(cid, &bitmap, dev);
+
+  return bitmap;
+}
+
+/****************************************************************************
+ * Name: get_evtbuffinst_withlock
+ ****************************************************************************/
+
+static FAR alt_evtbuf_inst_t *get_evtbuffinst_withlock(
+  FAR struct alt1250_dev_s *dev, uint16_t cid, uint8_t altver,
+  FAR uint64_t *bitmap)
+{
+  FAR alt_evtbuf_inst_t *inst = NULL;
+  FAR alt_evtbuf_inst_t *ret = NULL;
+
+  cid = cid_to_searchable(cid, altver);
+
+  if (cid == APICMDID_SOCK_SELECT)
+    {
+      ret = &dev->select_inst;
+
+      lock_evtbuffinst(ret, dev);
+
+      ret->outparam = dev->select_container->outparam;
+      ret->outparamlen = dev->select_container->outparamlen;
+
+      search_evtbufinst(cid, bitmap, dev);
+    }
+  else
+    {
+      inst = search_evtbufinst(cid, bitmap, dev);
+      if (inst)
+        {
+          lock_evtbuffinst(inst, dev);
+
+          if (inst->stat == ALTEVTBUF_ST_WRITABLE)
+            {
+              ret = inst;
+            }
+          else
+            {
+              unlock_evtbufinst(inst, dev);
+            }
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: write_restart_param
+ ****************************************************************************/
+
+static void write_restart_param(FAR void *outp[], FAR void *buff)
+{
+  FAR int *out_reason = (FAR int *)outp[0];
+  FAR int *in_reason = (FAR int *)buff;
+
+  *out_reason = *in_reason;
+}
+
+/****************************************************************************
+ * Name: pollnotify
+ ****************************************************************************/
+
+static void pollnotify(FAR struct alt1250_dev_s *dev)
+{
+  nxsem_wait_uninterruptible(&dev->pfdlock);
+
+  if (dev->pfd)
+    {
+      /* If poll() waits, notify  */
+
+      poll_notify(&dev->pfd, 1, POLLIN);
+    }
+
+  nxsem_post(&dev->pfdlock);
+}
+
+/****************************************************************************
+ * Name: get_composehdlr
+ ****************************************************************************/
+
+compose_handler_t get_composehdlr(uint32_t cmdid, FAR uint8_t *payload,
+  size_t size)
+{
+  compose_handler_t ret = NULL;
```

Comment:
```suggestion
  compose_handler_t ret;
```

Reply:
This is not needed since `ret = alt1250_composehdlr(cmdid);` is followed in the next line